### ISSUE TITLE
no-expression-statements: shouldn't flag yield statements

### ISSUE THREAD

**Issue by danielnixon** [eslint-functional/eslint-plugin-functional#567]
## Bug Report

For the purposes of the `no-expression-statements` rule, I _think_ a yield statement should be treated like a return statement, i.e., not flagged as a lint issue.

```ts
export function* mySaga() {
  // Flagged by functional/no-expression-statements
  yield "hello";

  // Not flagged
  return "hello";
}
```

### Expected behavior

`yield xyz` not flagged

### Actual behavior

`yield xyz` flagged

### Steps to reproduce

See example above.

### Proposed changes

Add a dedicated special case to [checkExpressionStatement](https://github.com/eslint-functional/eslint-plugin-functional/blob/main/src/rules/no-expression-statements.ts#L74) for `yield` statements.
